climada_petals/engine/supplychain.py

```diff
@@ -250,101 +503,144 @@ def calc_indirect_impact(self, io_approach='ghosh'):
+                    for i in range(len(impact.event_id))
+                ],
+                axis=1,
+            ).T.set_index(impact.event_id)
+
+        elif io_approach == "ghosh":
+            value_added = calc_v(self.mriot.Z, self.mriot.x)
+            degr_value_added = (
+                self.secs_stock_shock * value_added.values * self.conv_fac()
+            )
+
+            self.indir_prod_impt_mat = pd.concat(
+                [
+                    calc_x_from_G(self.inverse, degr_value_added.iloc[i])
+                    for i in range(len(impact.event_id))
+                ],
+                axis=1,
+            ).T.set_index(impact.event_id)
+
+        elif io_approach == "eeioa":
+            self.indir_prod_impt_mat = (
+                pd.DataFrame(
+                    self.secs_stock_shock.dot(self.inverse)
+                    * self.mriot.x.values.flatten()
+                )
+                * self.conv_fac()
+            )
+
+        self.indir_prod_impt_eai = self.indir_prod_impt_mat.T.dot(impact.frequency)
+
+    def calc_total_production_impacts(self, impact):
+        """Calculate total production impacts."""
+        self.tot_prod_impt_mat = self.dir_prod_impt_mat.add(self.indir_prod_impt_mat)
+        self.tot_prod_impt_eai = self.tot_prod_impt_mat.T.dot(impact.frequency)
+
+    def calc_production_impacts(
+        self,
+        impact,
+        exposure,
+        impacted_secs=None,
+        io_approach=None,
+        stock_to_prod_shock=None,
+    ):
+        """Calculate direct, indirect and total production impacts.
+
+        Parameters
+        ----------
+        impact : Impact
+            Impact object with stocks impacts.
+        exposure : Exposures
+            Exposures object for impact calculation.
+        impacted_secs : range, np.ndarray or list, optional
+            The directly affected sectors. If range or np.ndarray,
+            it contains the affected sectors' positions in the MRIOT.
+            If list, it contains the affected sectors' names in the MRIOT.
+        """
+
+        self.calc_secs_exp_imp_shock(exposure, impact, impacted_secs)
+
+        self.calc_direct_production_impacts(impact, stock_to_prod_shock)
+        self.calc_indirect_production_impacts(impact, io_approach)
+        self.calc_total_production_impacts(impact)
+
+    def calc_matrixes(self, io_approach):
+        """
+        Build technical coefficient and Leontief inverse matrixes (if Leontief approach)
+        or allocation coefficients and Ghosh matrixes (if Ghosh approach).
+        """
+
+        io_model = {
+            "leontief": (pymrio.calc_A, pymrio.calc_L),
+            "eeioa": (pymrio.calc_A, pymrio.calc_L),
+            "ghosh": (calc_B, calc_G),
+        }
+
+        coeff_func, inv_func = io_model[io_approach]
+
+        self.coeffs = coeff_func(self.mriot.Z, self.mriot.x)
+        self.inverse = inv_func(self.coeffs)
+
+    def conv_fac(self):
```

Comment:
Unnecessary abbreviation?
```suggestion
    def conversion_factor(self):
```